hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -200,18 +206,11 @@ private void adjustCryptoAllowance(BalanceChange change, AccountID ownerID) {
 	private void adjustNftAllowance(final BalanceChange change, final AccountID ownerID) {
 		final var allowanceId = FcTokenAllowanceId.from(
 				change.getToken().asEntityNum(), EntityNum.fromAccountId(change.getPayerID()));
-		final var nftAllowances = new TreeMap<>(
-				(Map<FcTokenAllowanceId, FcTokenAllowance>) accountsLedger.get(ownerID, NFT_ALLOWANCES));
-		final var currentAllowance = nftAllowances.get(allowanceId);
-		if (!currentAllowance.isApprovedForAll()) {
-			var mutableAllowanceList = new ArrayList<>(currentAllowance.getSerialNumbers());
-			mutableAllowanceList.remove(change.serialNo());
-			if (mutableAllowanceList.isEmpty()) {
-				nftAllowances.remove(allowanceId);
-			} else {
-				nftAllowances.put(allowanceId, FcTokenAllowance.from(mutableAllowanceList));
-			}
-			accountsLedger.set(ownerID, NFT_ALLOWANCES, nftAllowances);
+		final var approveForAllNftsAllowances = new TreeSet<>(
```

Comment:
Since this no longer needs to mutate the NFT allowances, no need to construct a new `TreeSet` here.